### PR TITLE
Add virtio-vsock integration tests

### PR DESCRIPTION
**Pull request by sboeuf**
This pull request ensures `virtio-vsock` is functional, relying on a new Clear Linux cloud image including the `socat` utility.

### CODE REVIEW

**Comment by sboeuf** [comment]
@sameo @rbradford PTAL

File: src/main.rs [hunk 1]
```diff
@@ -636,54 +678,16 @@ mod tests {
...
         }
 
         fn ssh_command_l1(&self, command: &str) -> Result<String, Error> {
-            self.ssh_command_ip(command, &self.network.guest_ip)
+            ssh_command_ip(command, &self.network.guest_ip)
         }
 
         fn ssh_command_l2(&self, command: &str) -> Result<String, Error> {
-            self.ssh_command_ip(command, &self.network.l2_guest_ip)
+            ssh_command_ip(command, &self.network.l2_guest_ip)
```
Comment (sameo) [inline src/main.rs]: Although I'm fine with the `self` removal, I would also move it into a different PR as it adds some noise to this PR.

File: scripts/run_integration_tests.sh [hunk 2]
```diff
@@ -135,7 +135,7 @@ sudo setcap cap_net_admin+ep /usr/bin/qemu-system-x86_64
 sudo adduser $USER kvm
 newgrp kvm << EOF
 export RUST_BACKTRACE=1
-cargo test --features "integration_tests"
+cargo test --features "integration_tests" -- --nocapture
```
Comment (sameo) [inline scripts/run_integration_tests.sh]: May we move that one to its own PR? Or do we really need it here?

Comment (rbradford) [reply scripts/run_integration_tests.sh]: I think the bigger issue is that we really need to redirect all console/serial output to a file and then print the full contents when the test completes (perhaps only do it on error)

Comment (sboeuf) [reply scripts/run_integration_tests.sh]: I can definitely move this one to another PR, and we can discuss there the details about console/serial redirection :)

File: src/main.rs (hunk 1 above)
Comment (sboeuf) [reply src/main.rs]: Okay let me send a separate PR for this.

**Comment by sboeuf** [comment]
@sameo PR updated! It depends on #266 to first be merged.
